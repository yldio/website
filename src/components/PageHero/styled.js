import styled from 'styled-components';
import { highlight, textLight } from 'styles/colours';
import { gridWidth, spacing } from 'styles/spacing';
import is from 'styled-is';
import remcalc from 'remcalc';

export const DefaultTitle = styled.h1`
  font-size: 3rem;
  font-weight: 500;
  color: ${textLight};
  line-height: 1.2em;
`;

export const Before = styled.h1`
  color: ${highlight};
  font-size: 1.125rem;
  font-weight: 700;
  margin-bottom: ${spacing()};
  text-transform: uppercase;

  ${is('white')`
    color: white;
  `};

  ${is('big')`
    text-transform: none;
    font-size: ${remcalc(72)};
  `};
`;

export const Header = styled.header`
  margin-top: ${spacing(4)};
  margin-bottom: ${spacing(2)};
`;

export const Wrapper = styled.div`
  max-width: ${gridWidth * 2 / 3}px;
`;

export const Main = styled.main`
  color: ${textLight};
  font-size: 1.25rem;
  line-height: 1.8em;

  ${is('white')`
    color: white;
    font-size: ${remcalc(18)};
  `};
`;
