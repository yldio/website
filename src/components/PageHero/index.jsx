import React from 'react';
import PropTypes from 'prop-types';
import { Before, Header, DefaultTitle, Main, Wrapper } from './styled';

const PageHero = ({ before, children, title, white, big }) => {
  const renderedTitle =
    typeof title === 'string' ? <DefaultTitle>{title}</DefaultTitle> : title;

  return (
    <Wrapper>
      <Header>
        {before && (
          <Before white={white} big={big}>
            {before}
          </Before>
        )}
        {renderedTitle}
      </Header>
      <Main white={white}>{children}</Main>
    </Wrapper>
  );
};

PageHero.propTypes = {
  before: PropTypes.string,
  children: PropTypes.node,
  white: PropTypes.bool,
  big: PropTypes.bool,
  title: PropTypes.node.isRequired
};

PageHero.defaultProps = {
  before: null,
  children: [],
  white: false,
  big: false
};

export { Section } from './styled';

export default PageHero;
