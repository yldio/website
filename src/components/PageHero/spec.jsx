import React from 'react';
import { shallow } from 'enzyme';
import PageHero from './index';

describe('components/PageHero', () => {
  test('basic snapshot', () => {
    const wrapper = shallow(
      <PageHero title="Hello, world!">I am the page hero body</PageHero>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('renders children', () => {
    const children = 'I am the page hero body';

    const wrapper = shallow(
      <PageHero title="Hello, world!">{children}</PageHero>
    );

    expect(
      wrapper
        .children()
        .find('styled__Main')
        .children()
        .text()
    ).toEqual(children);
  });

  test('renders DefaultTitle if only a string is provided as title', () => {
    const wrapper = shallow(<PageHero title="Hello, world!" />);

    expect(wrapper.find('styled__Header').contains('Hello, world!')).toBe(true);
  });

  test('renders title as a component if it is not a string', () => {
    const title = <div>Hello, world</div>;
    const wrapper = shallow(<PageHero title={title} />);

    expect(wrapper.find('styled__Header').contains(title)).toBe(true);
  });
});
