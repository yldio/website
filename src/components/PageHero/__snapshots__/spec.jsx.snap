// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PageHero basic snapshot 1`] = `
<styled__Wrapper>
  <styled__Header>
    <styled__DefaultTitle>
      Hello, world!
    </styled__DefaultTitle>
  </styled__Header>
  <styled__Main
    white={false}
  >
    I am the page hero body
  </styled__Main>
</styled__Wrapper>
`;
